Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -59,6 +60,17 @@ static String ak_string_from(VM& vm, GlobalObject& global_object)
     return Value(this_object).to_string(global_object);
 }
 
+static Optional<size_t> split_match(AK::String haystack, size_t start, AK::String needle)
```

Comment:
We're usually not using the `AK::` prefix (no idea what that is actually called :smile:), also I think those strings can be const references, no copy needed:

```suggestion
static Optional<size_t> split_match(const String& haystack, size_t start, const String& needle)
```